### DISCUSSION TITLE
Having issues with Paddle Billing webhooks

### DISCUSSION THREAD

**Discussion by chrismahon** [pay-rails/pay#931] [General]
Hi, I seem to be having issues with web hooks after successfully creating a subscription using the Paddle.js button example.

In my Rails logs I keep seeing this error, presumably to sync the subscription with my Rails app:

```ActionController::RoutingError (No route matches [POST] "/pay/webhooks/paddle_billing"):```

I have this setup in my `/initializers/pay.rb`:

```
config.automount_routes = true
config.routes_path = "/pay"
config.enabled_processors = [:paddle_billing]
```

Inside the Paddle Notifications screen I have setup this destination:

`https://mydomain.com/pay/webhooks/paddle_billing`

Is there something obvious I'm missing? I've tried going through the documentation to see if I've missed something but can't spot anything obvious.

When I run `rails routes` I see this:

```
pay         /pay         Pay::Engine

Routes for Pay::Engine:
payment GET  /payments/:id(.:format) pay/payments#show
```

**Comment by excid3** [comment]
The routes are included conditionally:

https://github.com/pay-rails/pay/blob/main/config/routes.rb#L7
https://github.com/pay-rails/pay/blob/main/lib/pay/paddle_billing.rb#L16-L20

**Comment by chrismahon** [comment]
Hey Chris, thanks for the reply – I have Paddle Billing setup in my `/initializers/pay.rb` file though?

```
config.automount_routes = true
config.routes_path = "/pay"
config.enabled_processors = [:paddle_billing]
```

Is there somewhere else I have to enable it?

**Reply by excid3** [reply]
See the second link I shared. It checks the config and verifies the gem is loaded.

**Reply by chrismahon** [reply]
Thanks Chris,

Not sure what's going on then because I have the paddle gem installed:

```
gem "pay", "~> 6.0"
gem "paddle", "~> 2.1"
gem "receipts", "~> 2.0"
```

Just checked in rails console to see if it's loaded as well:

```
irb(main):001> defined?(::Paddle)
=> "constant"
```

If I do a simple debug in my Rails view files as well both of these return as expected.

`<%= Pay.enabled_processors.include?(:paddle_billing) %>` returns `true`
`<%= defined?(::Paddle) %>` returns `constant`

Sorry if I'm missing something obvious Chris!

**Reply by excid3** [reply]
You're a major version behind on Pay there. Try upgrading to Pay 7.

**Reply by chrismahon** [reply]
Thanks Chris that was it!

I was following the [latest installation guide](https://github.com/pay-rails/pay/blob/main/docs/1_installation.md) which mentions 6.0:

`gem "pay", "~> 6.0"`

I've [created a PR](https://github.com/pay-rails/pay/pull/932) to update the readme docs if that's of any help.
